objects/src/notes/inputs.rs

```diff
@@ -70,28 +70,31 @@ impl NoteInputs {
         &self.values
     }
 
-    /// Returns a vector of input values padded with ZEROs to the next multiple of 8.
-    pub fn to_padded_values(&self) -> Vec<Felt> {
-        pad_inputs(&self.values)
-    }
-
-    /// Returns a vector of input values.
-    pub fn to_vec(&self) -> Vec<Felt> {
-        self.values.to_vec()
+    /// Returns the note's input formatted to be used with the advice map.
+    ///
+    /// The format is `input_len || INPUTS || PADDING`, where:
+    ///
+    /// - input_len is the number of inputs
+    /// - INPUTS is the variable inputs for the note
+    /// - PADDING is the optional padding to align the data with a 2WORD boundary
+    ///
+    pub fn format_for_advice(&self) -> Vec<Felt> {
+        // NOTE: keep map in sync with the `note::get_inputs` API procedure
+        let mut padded = pad_inputs(&self.values);
+        padded.insert(0, self.num_values().into());
+        padded
     }
+}
 
-    // UTILITIES
-    // --------------------------------------------------------------------------------------------
+impl Default for NoteInputs {
+    fn default() -> Self {
+        let values = vec![];
+        let hash = {
+            let padded_values = pad_inputs(&values);
+            Hasher::hash_elements(&padded_values)
+        };
```

Comment:
I opted to not due that because `new` returns a `Result`, whereas `default` doesnt

Reply:
Ah - I see! Should we then encapsulate the logic for computing inputs hash into a small helper function (so that we don't duplicate it here and in the constructor)?

Reply:
added